akka-remote/src/main/scala/akka/remote/artery/tcp/ConfigSSLEngineProvider.scala

```diff
@@ -42,14 +42,18 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
       system.settings.config.getConfig("akka.remote.artery.ssl.config-ssl-engine"),
       Logging.withMarker(system, classOf[ConfigSSLEngineProvider].getName))
 
+  val sslEngineConfig = new SSLEngineConfig(config)
```

Comment:
I'm tempted to make this one `protected` and deprecate all five fields below.

Reply:
I'd like to keep this one private for now, so we can evolve it more freely later.

If we make it public now it means we also need to 'get it right' which seems ambitious.